### DISCUSSION TITLE
with-apollo issues; dynamic content not rendering in Head in page source

### DISCUSSION THREAD

**Discussion by Segun98** [vercel/next.js#12292] [Help] [answered]
Dynamic contents are not rendering in Head in view page source, all meta tags are empty. 

That defeats the purpose on SEO in SSR please, help.

repo: https://github.com/Segun98/Blog-Nextjs/blob/master/client/pages/post/%5Bid%5D.jsx

Page source screenshot

![Screenshot (67)](https://user-images.githubusercontent.com/53624548/80537335-c79b4780-899b-11ea-9675-e8c2f4334b15.png)

Code. Content are from a Graphql query in a dyynamic page [id].js

![Screenshot (68)](https://user-images.githubusercontent.com/53624548/80537358-d2ee7300-899b-11ea-977f-d165cc1ed50d.png)

**Comment by rafaelalmeidatk** [comment, marked as answer]
You are fetching the post at runtime so the data is only available after you visit the page. Instead, you can fetch the post at build time with `getStaticProps`: https://nextjs.org/docs/basic-features/data-fetching#getstaticprops-static-generation

**Reply by Segun98** [reply]
How do I use getStaticprops with apollo graphql?

**Reply by rafaelalmeidatk** [reply]
I wouldn't use Apollo for getStaticProps because it is too heavy just for some requests, you can use fetch or https://github.com/prisma-labs/graphql-request

**Reply by Segun98** [reply]
Thanks. Graphql-request worked just fine

**Reply by brianlovin** [reply]
Because I'm stubborn, I ended up pursuing the `getStaticProps` + Apollo route. It's probably still smarter to avoid using it, as it introduced a lot of headache, but if you are interested feel free to poke at some of the code I used:

- https://github.com/brianlovin/brian-lovin-next/blob/master/src/graphql/services/apollo.ts
- https://github.com/brianlovin/brian-lovin-next/blob/master/src/components/withApollo/index.tsx
- https://github.com/brianlovin/brian-lovin-next/blob/master/src/pages/bookmarks.tsx
